engine/src/main/java/org/terasology/engine/logic/location/LocationChangedSystem.java

```diff
@@ -2,23 +2,48 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.engine.logic.location;
 
+import com.google.common.collect.Sets;
 import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.entity.lifecycleEvents.OnAddedComponent;
 import org.terasology.engine.entitySystem.entity.lifecycleEvents.OnChangedComponent;
 import org.terasology.engine.entitySystem.event.ReceiveEvent;
 import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
 import org.terasology.engine.entitySystem.systems.RegisterMode;
 import org.terasology.engine.entitySystem.systems.RegisterSystem;
+import org.terasology.engine.entitySystem.systems.UpdateSubscriberSystem;
+
+import java.util.Set;
 
 @RegisterSystem(RegisterMode.AUTHORITY)
-public class LocationChangedSystem extends BaseComponentSystem {
+public class LocationChangedSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+
+    private Set<EntityRef> process = Sets.newHashSet();
 
     @ReceiveEvent(components = {LocationComponent.class})
-    public void onItemUpdate(OnChangedComponent event, EntityRef entity) {
+    public void locationChanged(OnAddedComponent event, EntityRef entity) {
         LocationComponent lc = entity.getComponent(LocationComponent.class);
-        if (!lc.lastPosition.equals(lc.position) || !lc.lastRotation.equals(lc.rotation)) {
-            entity.send(new LocationChangedEvent(lc, lc.lastPosition, lc.lastRotation));
-            lc.lastPosition.set(lc.position);
-            lc.lastRotation.set(lc.rotation);
+        lc.isDirty = false;
+    }
+
+    @ReceiveEvent(components = {LocationComponent.class})
+    public void locationChanged(OnChangedComponent event, EntityRef entity) {
+        LocationComponent lc = entity.getComponent(LocationComponent.class);
+        if (lc.isDirty && (!lc.position.equals(lc.lastPosition) || !lc.rotation.equals(lc.lastRotation))) {
+            process.add(entity);
+        } else {
+            lc.isDirty = false;
+        }
+    }
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef entity : process) {
```

Comment:
Would be interesting to collect some metrics here on how many entities we typically process per tick. Could easily be added in a follow up PR, I guess.